platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/v2/internal/emergency/EmergencyLoggerImpl.java

```diff
@@ -0,0 +1,263 @@
+
+    /**
+     * The message that is used when the message is undefined.
+     */
+    private static final String UNDEFINED_MESSAGE = "UNDEFINED-MESSAGE";
+
+    /**
+     * The size of the queue that is used to store the log events.
+     */
+    private static final int LOG_EVENT_QUEUE_SIZE = 1000;
+
+    /**
+     * The name of the system property that defines the level of the logger.
+     */
+    private static final String LEVEL_PROPERTY_NAME = "com.swirlds.logging.emergency.level";
+
+    /**
+     * The singleton instance of the logger.
+     */
+    private static final EmergencyLoggerImpl INSTANCE = new EmergencyLoggerImpl();
+
+    /**
+     * The level that is supported by the logger.
+     */
+    private final Level supportedLevel;
+
+    /**
+     * The queue that is used to store the log events. Once the real logging system is available the events can be taken
+     * by the logging system and logged.
+     */
+    private final ArrayBlockingQueue<LogEvent> logEvents;
+
+    /**
+     * A thread local that is used to prevent recursion. This can happen when the logger is used in a broken system.
+     */
+    private final ThreadLocal<Boolean> recursionGuard;
+
+    /**
+     * Creates the singleton instance of the logger.
+     */
+    private EmergencyLoggerImpl() {
+        this.logEvents = new ArrayBlockingQueue<>(LOG_EVENT_QUEUE_SIZE);
+        recursionGuard = new ThreadLocal<>();
+        supportedLevel = getSupportedLevelFromSystemProperties();
+    }
+
+    /**
+     * Returns the level based on a possible system property.
+     *
+     * @return the level based on a possible system property
+     */
+    @NonNull
+    private static Level getSupportedLevelFromSystemProperties() {
+        final String property = System.getProperty(LEVEL_PROPERTY_NAME);
+        if (property == null) {
+            return Level.DEBUG; // DEFAULT LEVEL
+        } else if (Objects.equals(property.toUpperCase(), Level.TRACE.name())) {
+            return Level.TRACE;
+        } else if (Objects.equals(property.toUpperCase(), Level.DEBUG.name())) {
+            return Level.DEBUG;
+        } else if (Objects.equals(property.toUpperCase(), Level.INFO.name())) {
+            return Level.INFO;
+        } else if (Objects.equals(property.toUpperCase(), Level.WARN.name())) {
+            return Level.WARN;
+        } else if (Objects.equals(property.toUpperCase(), Level.ERROR.name())) {
+            return Level.ERROR;
+        } else {
+            return Level.TRACE;
+        }
+    }
+
+    @Override
+    public void logNPE(@NonNull final String nameOfNullParam) {
+        log(
+                Level.ERROR,
+                "Null parameter: " + nameOfNullParam,
+                new NullPointerException("Null parameter: " + nameOfNullParam));
+    }
+
+    @Override
+    public void log(@NonNull final Level level, @NonNull final String message, @Nullable final Throwable thrown) {
+        if (level == null && message == null) {
```

Comment:
level and message as marked with `@NonNull`, then checked for `null`

Reply:
The annotation does not forbid to call it with null. It is only checked in sonar.